pkg/arvo/sur/graph-store.hoon

```diff
@@ -0,0 +1,69 @@
+/-  *post
+|%
+++  mop
+  |*  [key=mold value=mold]
+  |=  ord=$-([key key] ?)
+  |=  a=*
+  =/  b  ;;((tree [key=key val=value]) a)
+  ?>  (check-balance:((ordered-map key value) ord) b)
+  b
+::
++$  mark          term
++$  graph         ((mop atom node) gth)
++$  marked-graph  [p=graph q=(unit mark)]
+::
++$  node          [=post children=internal-graph]
++$  graphs        (map resource marked-graph)
+::
++$  tag-queries   (jug term resource)
+::
++$  update-log    ((mop time logged-update) gth)
++$  update-logs   (map resource update-log)
+::
++$  internal-graph
+  $~  [%empty ~]
+  $%  [%graph p=graph]
+      [%empty ~]
+  ==
+::
++$  network
+  $:  =graphs
+      =tag-queries
+      =update-logs
+      archive=graphs
+  ==
+::
++$  update
+  $%  [%0 p=time q=update-0]
+  ==
+::
++$  logged-update
+  $%  [%0 p=time q=logged-update-0]
```

Comment:
Is the `p=time` in this data structure the same as the key in `$update-log`?  If so, we should remove it, since there's no point in a map item's value duplicating its key.  If not, a comment indicating what this time represents would be helpful.

Reply:
Yes, the poke and the value stored in the `update-log` are the same. To make them different ends up as a messier implementation, as we need some way to `poke` the store with the key-value pair.